crates/context/interface/src/transaction.rs

```diff
@@ -210,4 +211,16 @@ pub trait Transaction {
 
         Ok(effective_balance_spending)
     }
+
+    /// Returns the effective balance calculated with [`Self::effective_balance_spending`] but without the value.
+    #[inline]
+    fn effective_balance_spending_without_value(
+        &self,
+        base_fee: u128,
+        blob_price: u128,
+    ) -> Result<U256, InvalidTransaction> {
+        Ok(self
+            .effective_balance_spending(base_fee, blob_price)?
+            .saturating_sub(self.value()))
```

Comment:
this now uses saturating sub, which seems okay because this is expected to be called after the max_balance_spending check but we could highlight this perhaps?